pkg/applicationserver/io/packages/loragls/v3/package.go

```diff
@@ -221,6 +220,9 @@ func (p *GeolocationPackage) pushUplink(
 		data.RecentMetadata = assocData.RecentMetadata
 
 		st, err := assocData.Struct()
+		if errors.Is(err, ErrEmptyData) {
```

Comment:
And here I would just check for `nil` `st` - can even be merged into the `err != nil` `if`.